src/analyses/static_analysis.h

```diff
@@ -86,6 +86,9 @@ class domain_baset
   // this computes the join between "this" and "b"
   // return true if "this" has changed
 
+  bool has_been_seen() const { return seen; }
+  void set_seen() { seen=true; }
```

Comment:
Putting aside the contextual discussion: could those changes be synchronised to ai.h as well, please?